packages/browser-agent-core/common/window/session-storage.js

```diff
@@ -0,0 +1,27 @@
+/**
+ * Interface for Agents' usage of the Window.sessionStorage as a replacement to third-party cookies.
+ *  (All agents on a tab|window will share the same sessionStorage.)
+ * 
+ * @documentation https://newrelic.atlassian.net/wiki/spaces/INST/pages/2522513791/JSESSIONID+Cookie+Change+Design+Document
+ * @environment Browser script
+ */
+import { generateRandomHexString } from '../ids/unique-id'
+
+export { getCurrentSessionId };
+
+const ss = window.sessionStorage;
+const SESS_ID = Symbol("session id").toString(); // prevents potential key collisions in session storage
+
+/**
+ * @returns {string} This tab|window's session identifier, or a new ID if not found in storage
+ */
+function getCurrentSessionId() {
```

Comment:
I like this method, but can we change the name a bit to illustrate what it is doing more succinctly?

something like `getOrSetCurrentSessionId()` or whatever you like that illustrates that it is also setting state in certain circumstances.

Reply:
Revised